x-pack/plugins/security_solution/public/common/components/endpoint/host_isolation/from_alerts/use_host_isolation_action.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback, useMemo } from 'react';
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import {
+  HOST_ENDPOINT_UNENROLLED_TOOLTIP,
+  LOADING_ENDPOINT_DATA_TOOLTIP,
+  NOT_FROM_ENDPOINT_HOST_TOOLTIP,
+} from '../../responder';
+import { useAlertResponseActionsSupport } from '../../../../hooks/endpoint/use_alert_response_actions_support';
+import { useIsExperimentalFeatureEnabled } from '../../../../hooks/use_experimental_features';
+import type { AgentStatusInfo } from '../../../../../../common/endpoint/types';
+import { HostStatus } from '../../../../../../common/endpoint/types';
+import { useEndpointHostIsolationStatus } from './use_host_isolation_status';
+import { ISOLATE_HOST, UNISOLATE_HOST } from './translations';
+import { useUserPrivileges } from '../../../user_privileges';
+import type { AlertTableContextMenuItem } from '../../../../../detections/components/alerts_table/types';
+import { useAgentStatusHook } from '../../../../../management/hooks/agents/use_get_agent_status';
+
+interface UseHostIsolationActionProps {
+  closePopover: () => void;
+  detailsData: TimelineEventsDetailsItem[] | null;
+  isHostIsolationPanelOpen: boolean;
+  onAddIsolationStatusClick: (action: 'isolateHost' | 'unisolateHost') => void;
+}
+
+export const useHostIsolationAction = ({
+  closePopover,
+  detailsData,
+  isHostIsolationPanelOpen,
+  onAddIsolationStatusClick,
+}: UseHostIsolationActionProps): AlertTableContextMenuItem[] => {
+  const {
+    isSupported: hostSupportsResponseActions,
+    isAlert,
+    unsupportedReason,
+    details: {
+      agentType,
+      agentId,
+      agentSupport: { isolate: isolationSupported },
+    },
+  } = useAlertResponseActionsSupport(detailsData);
+  const agentStatusClientEnabled = useIsExperimentalFeatureEnabled('agentStatusClientEnabled');
+  const useAgentStatus = useAgentStatusHook();
+  const { canIsolateHost, canUnIsolateHost } = useUserPrivileges().endpointPrivileges;
+
+  const isEndpointAgent = useMemo(() => {
+    return agentType === 'endpoint';
+  }, [agentType]);
+
+  const {
+    loading: loadingHostIsolationStatus,
+    isIsolated,
+    agentStatus,
+    capabilities,
+  } = useEndpointHostIsolationStatus({
+    agentId,
+    agentType,
+  });
+
+  const { data: externalAgentData } = useAgentStatus([agentId], agentType, {
```

Comment:
I'm not sure I follow... 
The `useAgentStatusHook()` already does feature flag checks and the crowdstrike functionality expect/requires that the `agentStatusClientEnabled` be enabled.


_(also: remind me to talk to you about this "hook-that-return-another-hook"... Its not actually behaving like a hook in this code here and likely should not be something you should do going forward. I left it as is only because we're getting ready to enable the FF for `agentStatusClientEnabled` anyway)_